src/file/file.service.ts

```diff
@@ -249,32 +246,37 @@ export class FileService {
   }
 
   /**
-   * Auxillary function for getting the nested array of files for getDescendantsByFolder.
+   * Retrieves the populated folder with its children, recursively, as a nested object.
+   * @param folderID - the ancestor folder.
+   * @param ownerID - owner of said folder.
+   * @param query - the conditions.
+   */
+  private static async getPopulatedChildren(folderID: string, ownerID: string, query: Partial<IFile>) : Promise<ResFile> {
+    const ancestor: ResFile = new ResFile(await this.getById(folderID));
+    const children: ResFile[] = await this.getPopulatedChildren_aux(folderID, ownerID, query);
+    ancestor.children = children;
+    return ancestor;
+  }
+
+  /**
+   * Auxillary function for getting the nested array of children files for getPopulatedChildren.
    * @param folderID - the ancestor folder.
    * @param ownerID - owner of said folder.
    * @param query - the conditions.
    * @param currArray - the array sent in the recursion.
    */
-  private static async getPopulatedChildren(folderID: string, ownerID: string, query: object, file: Partial<ResFile>) : Promise<ResFile[]> {
+  private static async getPopulatedChildren_aux(folderID: string, ownerID: string, query: Partial<IFile>) : Promise<ResFile[]> {
     const folderFiles: IFile[] = await this.getFilesByFolder(folderID, ownerID, query);
-    const resFolderFiles: ResFile[] = [];
-    const currFile : Partial<ResFile> = {};
+    const childrenArray : ResFile[] = [];
 
-    // extract essential properties from file
-    for (const prop in file) {
-      if (file[prop]) {
-        currFile[prop] = file[prop];
-      }
-    }
     // use the recursion to get the file's populated children
-    currFile.children = [];
     for (let i = 0 ; i < folderFiles.length ; i++) {
```

Comment:
Not that important but still noteworthy, you can use `folderFiles.forEach()`, it has more JS vibe to it.

Reply:
agreed. done.